### ISSUE TITLE
Add support for creating standalone VMs

### ISSUE THREAD

**Issue by nilic** [vmware/terraform-provider-vcd#399]
### Terraform Version
Terraform v0.12.12
provider.vcd v2.5.0

### Affected Resource(s)
- vcd_vapp_vm

### Actual Behavior
_vcd_vapp_vm_ resource has a _vapp_name_ as a required argument, which means that every VM created using the provider must be a part of a vApp container.

### Expected Behavior
Since vCD 9.0, it is possible to create standalone VMs, ie. VMs which are not a part of a vApp. It would be preferable if the Terraform vCD provider would also allow for creating standalone VMs, either through a new resource type (since the current one is called "vapp_vm") or by allowing _vapp_name_ argument of the _vcd_vapp_vm_ to be optional.

### References
[vCloud Director 9.0: Digging into the new standalone VM feature](https://anthonyspiteri.net/vcloud-director-9-0-digging-into-the-new-standalone-vm-feature/) - an overview of the standalone VM feature both from the user perspective and in the backend.

**Comment by dataclouder** [comment]
Just to be clear, we ara talking about a different resource here. What we have now is a resource representing a VM within a vApp (hence the name `vcd_vapp_vm`).
Our SDK doesn't support standalone VMs yet.

We don't need to change behavior of the current resource, but to create a separate one.

**Comment by lvirbalas** [comment]
Thanks for a good link with the feature request!

p.s. And, yes, we do recognize where your avatar is from :)

**Comment by rejanu** [comment]
I'd like to see this being added asap.

**Comment by Biibe** [comment]
This "vcd_standalone_vm" would be very useful!

**Comment by vbauzysvmware** [comment]
In reality that functionality still creates vApp and you can found it in UI just by changing default filtering settings. So there isn't standalone VM. Ui just generates vApp with random name and hides it in UI.

**Comment by reski-rukmantiyo** [comment]
Hi @vbauzysvmware , i'm interested in this use case.
does standalone vm is the same like you have same vApp and VM name?
Does this project will accomodate this feature in 2.9?

**Comment by vbauzysvmware** [comment]
Hi @reski-rukmantiyo 

At the moment we believe that it’s against Terraform practices to link real VM resource to a hidden vApp implicitly, because it induces a cause for state synchronization issues, which may lead to unwanted problems. 

![image](https://user-images.githubusercontent.com/41415459/84635393-d1e9b480-aefb-11ea-8bb0-fb74e6b2aa44.png)

If you faced good example in other Terraform providers where similar situation is done, please share with us.

**Comment by lvirbalas** [comment]
What we could do is make `vapp_name` optional and create a VM without vApp. Would be very simple for the user.

**Comment by jpbuecken** [comment]
> What we could do is make vapp_name optional and create a VM without vApp. Would be very simple for the user.

Nice idea. What will be the behaviour if someone creates a VM without vapp_name. Then later, he adds the flag vapp_name.

I can think of at least 2 solutions:
- Force recreation
- Make use of the "Move to" feature of VCD (online)

The second behavior will improve step 2 in comment https://github.com/vmware/terraform-provider-vcd/issues/579#issuecomment-714231196 since you can do it via terraform. And you need to modify the hcl anyway.

For the opposite: I'm not sure if it is possible to move a vm out of a vapp and keep it running as standalone vm.

**Comment by dataclouder** [comment]
Released in [3.2.0](https://github.com/vmware/terraform-provider-vcd/releases/tag/v3.2.0)
See a full example of usage [here](https://gist.github.com/dataclouder/20bea04eeb59c4dc821fc40edf426c0c)

Documentation:
*  [resource](https://github.com/vmware/terraform-provider-vcd/blob/master/website/docs/r/vm.html.markdown) 
* [data source](https://github.com/vmware/terraform-provider-vcd/blob/master/website/docs/d/vm.html.markdown).

A quick benchmark:
```
1 vApp with 10 VMs

          with network     bare VM
          ------------     -----------
Apply     21m  5.097s      12m 24.084s
Destroy    5m 47.492s       4m 26.281s

10 standalone VM

          with network     bare VM
          ------------     -----------
Apply      2m 58.690s       1m 31.818s
Destroy    1m 15.381s       0m 40.361s
```

Note: times may be a lot better or a lot worse depending on the hardware + network used.

You can quickly try the multiple VM creation with the trick below.

```hcl
resource "vcd_vm" "TestMultiStandalone" {
  for_each      = toset([ "one", "two", "three", "four", "five", "six", "seven", "eight", "nine", "ten"])
  org           = "datacloud"
  vdc           = "vdc-datacloud"
  name          = each.value
  catalog_name  = "cat-datacloud"
  template_name = "photon-hw11"
  memory        = 1024
  cpus          = 1
  power_on      = false
}
```
and the corresponding vApp+VM
```hcl

resource "vcd_vapp" "multiVmVapp" {
  org  = "datacloud"
  vdc  = "vdc-datacloud"
  name = "multiVmVapp"
  power_on = false
}
resource "vcd_vapp_vm" "testMultiVappVm" {
  for_each      = toset([ "one", "two", "three", "four", "five", "six", "seven", "eight", "nine", "ten" ])
  org           = "datacloud"
  vdc           = "vdc-datacloud"
  vapp_name     = "multiVmVapp"
  name          = each.value
  catalog_name  = "cat-datacloud"
  template_name = "photon-hw11"
  memory        = 1024
  cpus          = 1
  power_on      = false
}
```
